src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -49,7 +56,30 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
   const stakingOpportunitiesById = useSelector(selectStakingOpportunitiesById)
 
   const maybeRefetchOpportunities = useCallback(
-    ({ chainId, transfers }: Transaction, accountId: AccountId) => {
+    ({ chainId, transfers, status }: Transaction, accountId: AccountId) => {
+      if (status === TxStatus.Confirmed) return
+
+      if (
+        isSupportedThorchainSaversChainId(chainId) &&
+        transfers.some(({ assetId }) => isSupportedThorchainSaversAssetId(assetId))
+      ) {
+        const { getOpportunitiesUserData } = opportunitiesApi.endpoints
+        // Artificial longer completion time, since THORChain Txs take around 15s after confirmation to be picked in the API
+        // This way, we ensure "View Position" actually routes to the updated position
+        new Promise(resolve => setTimeout(resolve, 17000)).then(() => {
```

Comment:
how do you feel about a lil util fn so we can `await sleep(17000)` like we're python kids?

Reply:
[here_we_go](9d88bed8e147bae9f75315e13f61b708f9e70cc0)